File: src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/AgencyInternalLocationProfileRepositoryTest.java
package uk.gov.justice.hmpps.prison.repository.jpa.repository;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.test.context.ActiveProfiles;
import uk.gov.justice.hmpps.prison.repository.jpa.model.AgencyInternalLocationProfile;
import uk.gov.justice.hmpps.prison.repository.jpa.model.HousingAttributeReferenceCode;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;
import static org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase.Replace.NONE;

@DataJpaTest
@ActiveProfiles("test")

@AutoConfigureTestDatabase(replace = NONE)
public class AgencyInternalLocationProfileRepositoryTest {

    @Autowired
    private AgencyInternalLocationProfileRepository repository;

    @Test
    public void findAllByLivingUnitAndAgencyIdAndDescription() {
        final var expected = List.of(
                                AgencyInternalLocationProfile.builder()
                                    .locationId(-3L)
                                    .code("DO")
                                    .profileType("HOU_UNIT_ATT")
                                    .housingAttributeReferenceCode(new HousingAttributeReferenceCode("DO", "Double Occupancy"))
                                    .build(),
                                AgencyInternalLocationProfile.builder()
                                    .locationId(-3L)
                                    .code("LC")
                                    .profileType("HOU_UNIT_ATT")
                                    .housingAttributeReferenceCode(new HousingAttributeReferenceCode("LC", "Listener Cell"))
                                    .build());

        final var profiles = repository.findAllByLocationId(-3L);

        assertThat(profiles).isEqualTo(expected);
    }

}

File: src/test/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/AgencyInternalLocationProfileRepositoryTest.kt
package uk.gov.justice.hmpps.prison.repository.jpa.repository

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import org.springframework.test.context.ActiveProfiles
import uk.gov.justice.hmpps.prison.repository.jpa.model.AgencyInternalLocationProfile
import uk.gov.justice.hmpps.prison.repository.jpa.model.HousingAttributeReferenceCode

@DataJpaTest
@ActiveProfiles("test")
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
class AgencyInternalLocationProfileRepositoryTest {
  @Autowired
  private lateinit var repository: AgencyInternalLocationProfileRepository

  @Test
  fun findAllByLivingUnitAndAgencyIdAndDescription() {
    val expected = listOf(
      AgencyInternalLocationProfile.builder()
        .locationId(-3L)
        .code("DO")
        .profileType("HOU_UNIT_ATT")
        .housingAttributeReferenceCode(HousingAttributeReferenceCode("DO", "Double Occupancy"))
        .build(),
      AgencyInternalLocationProfile.builder()
        .locationId(-3L)
        .code("LC")
        .profileType("HOU_UNIT_ATT")
        .housingAttributeReferenceCode(HousingAttributeReferenceCode("LC", "Listener Cell"))
        .build(),
    )
    val profiles = repository.findAllByLocationId(-3L)
    assertThat(profiles).isEqualTo(expected)
  }
}
